sdk/internal/mock/mock.go

```diff
@@ -58,6 +60,27 @@ func NewTLSServer() (*Server, func()) {
 	return s, func() { s.srv.Close() }
 }
 
+// NewTLSHTTP2Server creates a new Server object with HTTP2 configuration.
+// The returned close func must be called when the server is no longer needed.
+// It will return nil for both the server and close func if it encountered an error
+// when configuring HTTP2 for the new TLS server.
+func NewTLSHTTP2Server() (*Server, func()) {
```

Comment:
Instead of adding a new constructor, how about making the existing ones use variadic configuration?  It seems likely we'll want to add more configuration options in the future.

Reply:
Yeah makes sense, let me look into it.

Reply:
Done